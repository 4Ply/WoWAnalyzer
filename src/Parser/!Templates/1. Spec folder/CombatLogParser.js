import CoreCombatLogParser from 'Parser/Core/CombatLogParser';

// TODO: Pick the ones relevant to your spec:
// import DamageDone from 'Parser/Core/Modules/DamageDone';
// import HealingDone from 'Parser/Core/Modules/HealingDone';
// import DamageTaken from 'Parser/Core/Modules/DamageTaken';

import Abilities from './Modules/Abilities';
import Checklist from './Modules/Features/Checklist';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';

// TODO: import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // TODO: Enable the ones relevant to your spec:
    // damageDone: [DamageDone, { showStatistic: true }],
    // healingDone: [HealingDone, { showStatistic: true }],
    // damageTaken: [DamageTaken, { showStatistic: true }],

    abilities: Abilities, // TODO: Go to Abilities class and fill it with all abilities available to your spec

    // Features
    checklist: Checklist, // TODO: Go into Checklist to define rules
    alwaysBeCasting: AlwaysBeCasting, // TODO: Define a list of all abilities on the GCD
    // TODO: cooldownThroughputTracker: CooldownThroughputTracker,
  };
}

export default CombatLogParser;
