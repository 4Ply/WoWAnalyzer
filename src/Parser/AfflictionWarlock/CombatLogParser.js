import React from 'react';

import SuggestionsTab from 'Main/SuggestionsTab';
import Tab from 'Main/Tab';
import Talents from 'Main/Talents';

import MainCombatLogParser from 'Parser/Core/CombatLogParser';

import CastEfficiency from './Modules/Features/CastEfficiency';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownTracker from './Modules/Features/CooldownTracker';
import AgonyUptime from './Modules/Features/AgonyUptime';
import CorruptionUptime from "./Modules/Features/CorruptionUptime";
import SiphonLifeUptime from "./Modules/Features/SiphonLifeUptime";

import SoulShardTracker from "./Modules/SoulShards/SoulShardTracker";
import SoulShardDetails from "./Modules/SoulShards/SoulShardDetails";

class CombatLogParser extends MainCombatLogParser {

  static specModules = {
    // Features
    castEfficiency: CastEfficiency,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownTracker: CooldownTracker,

    // DoT uptimes
    agonyUptime: AgonyUptime,
    corruptionUptime: CorruptionUptime,
    siphonLifeUptime: SiphonLifeUptime,

    //Core
    soulShardTracker: SoulShardTracker,
    soulShardDetails: SoulShardDetails,
  };

  generateResults() {
    const results = super.generateResults();
    results.tabs = [
      {
        title: 'Suggestions',
        url: 'suggestions',
        render: () => (
          <SuggestionsTab issues={results.issues} />
        ),
      },
      {
        title: 'Talents',
        url: 'talents',
        render: () => (
          <Tab title="Talents">
            <Talents combatant={this.selectedCombatant} />
          </Tab>
        ),
      },
      // {
      //   title: 'Mana',
      //   url: 'mana',
      //   render: () => (
      //     <Tab title="Mana" style={{ padding: '15px 22px' }}>
      //       <Mana
      //         reportCode={this.report.code}
      //         actorId={this.playerId}
      //         start={this.fight.start_time}
      //         end={this.fight.end_time}
      //       />
      //     </Tab>
      //   ),
      // },
      ...results.tabs,
    ];

    return results;
  }
}

export default CombatLogParser;
