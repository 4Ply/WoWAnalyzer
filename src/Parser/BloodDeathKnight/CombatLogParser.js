import React from 'react';
import SuggestionsTab from 'Main/SuggestionsTab';
import Tab from 'Main/Tab';
import Talents from 'Main/Talents';
import CoreCombatLogParser from 'Parser/Core/CombatLogParser';

import CastEfficiency from './Modules/Features/CastEfficiency';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import CooldownTracker from './Modules/Features/CooldownTracker';
import BloodPlagueUptime from './Modules/Features/BloodPlagueUptime';
import BoneShieldUptime from './Modules/Features/BoneShieldUptime';
import OssuaryUptime from './Modules/Features/OssuaryUptime';
import WastedDeathAndDecay from './Modules/Features/WastedDeathAndDecay';
import RPWasted from './Modules/Features/RunicPowerWasted';
import BlooddrinkerTicks from './Modules/Features/BlooddrinkerTicks';

class CombatLogParser extends CoreCombatLogParser {

  static specModules = {

    //DeathKnight Core

    // Features
    castEfficiency: CastEfficiency,
    alwaysBeCasting: AlwaysBeCasting,
    cooldownTracker: CooldownTracker,
    BoneShieldUptime: BoneShieldUptime,
    OssuaryUptime: OssuaryUptime,
    WastedDeathAndDecay: WastedDeathAndDecay,
    RunicPowerWasted: RPWasted,
    BlooddrinkerTicks: BlooddrinkerTicks,

    //DOT
    bloodplagueUptime: BloodPlagueUptime,
    // Talents


    // Traits


    // Items:

  };

  generateResults() {
    const results = super.generateResults();

    results.tabs = [
      {
        title: 'Suggestions',
        url: 'suggestions',
        render: () => (
          <SuggestionsTab issues={results.issues} />
        ),
      },
      {
        title: 'Talents',
        url: 'talents',
        render: () => (
          <Tab title="Talents">
            <Talents combatant={this.modules.combatants.selected} />
          </Tab>
        ),
      },
      ...results.tabs,
    ];

    return results;
  }
}

export default CombatLogParser;
