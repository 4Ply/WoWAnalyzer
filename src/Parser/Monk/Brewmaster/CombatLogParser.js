import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
import DamageDone from 'Parser/Core/Modules/DamageDone';
// Core
import HealingDone from './Modules/Core/HealingDone';
import DamageTaken from './Modules/Core/DamageTaken';
import HealingReceived from './Modules/Core/HealingReceived';
import Stagger from './Modules/Core/Stagger';
import StaggerFabricator from './Modules/Core/StaggerFabricator';
// Spells
import IronSkinBrew from './Modules/Spells/IronSkinBrew';
import BlackoutCombo from './Modules/Spells/BlackoutCombo';
// Features
import Checklist from './Modules/Features/Checklist';
import Abilities from './Modules/Features/Abilities';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
// Items
import T20_2pc from './Modules/Items/T20_2pc';
import T20_4pc from './Modules/Items/T20_4pc';
// normalizers
import IronskinBrewNormalizer from './Modules/Normalizers/IronskinBrew';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core
    healingDone: [HealingDone, { showStatistic: true }],
    healingReceived: HealingReceived,
    damageTaken: [DamageTaken, { showStatistic: true }],
    stagger: Stagger,
    staggerFabricator: StaggerFabricator,
    damageDone: [DamageDone, { showStatistic: true }],

    // Features
    checklist: Checklist,
    alwaysBeCasting: AlwaysBeCasting,
    abilities: Abilities,

    // Spells
    ironSkinBrew: IronSkinBrew,
    blackoutCombo: BlackoutCombo,

    // Items
    t20_2pc: T20_2pc,
    t20_4pc: T20_4pc,

    // normalizers
    isbNormalizer: IronskinBrewNormalizer,
  };
}

export default CombatLogParser;
