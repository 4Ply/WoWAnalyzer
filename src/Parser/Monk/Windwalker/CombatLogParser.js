import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
// Features
import DamageDone from 'Parser/Core/Modules/DamageDone';
import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
import Abilities from './Modules/Features/Abilities';
import CooldownThroughputTracker from './Modules/Features/CooldownThroughputTracker';
import Checklist from './Modules/Features/Checklist';
// Chi
import ChiDetails from './Modules/Chi/ChiDetails';
import ChiTracker from './Modules/Chi/ChiTracker';
// Core
import Channeling from './Modules/Core/Channeling';
// Spells
import ComboBreaker from './Modules/Spells/ComboBreaker';
import StormEarthAndFire from './Modules/Spells/StormEarthAndFire';
import FistsofFury from './Modules/Spells/FistsofFury';
import SpinningCraneKick from './Modules/Spells/SpinningCraneKick';
import ComboStrikes from './Modules/Spells/ComboStrikes';
import TouchOfKarma from './Modules/Spells/TouchOfKarma';
import BlackoutKick from './Modules/Spells/BlackoutKick';
// Talents
import HitCombo from './Modules/Talents/HitCombo';
import EnergizingElixir from './Modules/Talents/EnergizingElixir';

class CombatLogParser extends CoreCombatLogParser {
  static specModules = {
    // Core
    channeling: Channeling,

    // Features
    damageDone: [DamageDone, { showStatistic: true }],
    alwaysBeCasting: AlwaysBeCasting,
    abilities: Abilities,
    cooldownThroughputTracker: CooldownThroughputTracker,
    checklist: Checklist,

    // Resources
    chiTracker: ChiTracker,
    chiDetails: ChiDetails,

    // Talents:
    hitCombo: HitCombo,
    energizingElixir: EnergizingElixir,

    // Spells;
    comboBreaker: ComboBreaker,
    stormEarthAndFire: StormEarthAndFire,
    fistsofFury: FistsofFury,
    spinningCraneKick: SpinningCraneKick,
    touchOfKarma: TouchOfKarma,
    comboStrikes: ComboStrikes,
    blackoutKick: BlackoutKick,
  };
}

export default CombatLogParser;
