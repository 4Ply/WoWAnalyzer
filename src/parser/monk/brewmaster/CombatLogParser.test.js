import CombatLogParser from './CombatLogParser';
import integrationTest from 'parser/core/tests/CombatLogParserIntegrationTest';

const FIGHT_ID = 12;
const PLAYER_ID = 3;

describe('The Brewmaster Analyzer', integrationTest(
  CombatLogParser,
  __dirname + '/test-fixtures/example/meta.json',
  __dirname + '/test-fixtures/example/combatant-info.json',
  __dirname + '/test-fixtures/example/events.json.gz',
  FIGHT_ID,
  PLAYER_ID
));
