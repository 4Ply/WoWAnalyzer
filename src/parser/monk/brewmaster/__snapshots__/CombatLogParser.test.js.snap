// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Brewmaster Analyzer analyzers Abilities matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers AlwaysBeCasting matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt=""
            className="icon"
            src="//render-us.worldofwarcraft.com/icons/56/spell_mage_altertime.jpg"
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Downtime
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Downtime is available time not used to cast anything (including not having your GCD rolling). This can be caused by delays between casting spells, latency, cast interrupting or just simply not casting anything (e.g. due to movement/stunned).<br/>
        <li>You spent <b>58.39%</b> of your time casting something.</li>
        <li>You spent <b>41.61%</b> of your time casting nothing at all.</li>
        "
            >
              41.61 %
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
      >
        <div
          className="stat-health-bg"
          data-tip="You spent <b>58.39%</b> of your time casting something."
          style={
            Object {
              "width": "58.38917441936113%",
            }
          }
        >
          <img
            alt="Active time"
            src="/img/sword.png"
          />
        </div>
        <div
          className="remainder DeathKnight-bg"
          data-tip="You spent <b>41.61%</b> of your time casting nothing at all."
        >
          <img
            alt="Downtime"
            src="/img/afk.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers BlackOxBrew matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers BlackoutCombo matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers BreathOfFire matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers BrewCDR matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=100780"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_monk_tigerpalm.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Effective Brew CDR
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Your cooldowns were reduced by: <ul>
              <li>31 Keg Smash casts— <b>105.59s</b> (<b>18.41s</b> wasted)</li>
              
              <li>66 Tiger Palm hits — <b>57.00s</b> (<b>9.00s</b> wasted)</li>
              
            </ul>
            <b>Total cooldown reduction:</b> 162.59s.</b><br/>
            <b>Minimum Cooldown Reduction for 100% ISB uptime:</b> 25.88%"
            >
              34.59%
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers CelestialFortune matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=216521"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/inv_celestialserpentmount.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Celestial Fortune Healing
          </div>
          <div
            className="value"
          >
            1,982 HPS
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers Channeling matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers Checklist matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers DamageDone matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Sword"
            src="/img/sword.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Damage done
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Total damage done: <b>1,711,282</b> "
            >
              5,567 DPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers DamageTaken matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Shield"
            src="/img/shield.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Damage taken
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="The total damage taken was 4,244,362 (0 overkill)."
            >
              14k DTPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
        data-tip="Damage taken by type:
      <ul>
      <li><b>Physical</b>: 2,075,386 (48.90%)</li>
      <li><b>Magic</b>: 2,168,976 (51.10%)</li>
      </ul>
      Damage taken by magic school:
      <ul>
        <li><b>Physical</b>: 2,075,386 (48.90%)</li><li><b>Nature</b>: 1,856,517 (43.74%)</li><li><b>Shadow</b>: 312,459 (7.36%)</li>
      </ul>
      Click the bar to switch between simple and detailed mode."
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <div
          className="spell-school-1-bg"
          style={
            Object {
              "width": "48.89747858453167%",
            }
          }
        />
        <div
          className="spell-school-32-bg"
          style={
            Object {
              "width": "51.102521415468324%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers ElusiveFootwork matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers FitToBurst matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers GarothiWorldbreakerMelee matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers GiftOfTheOx matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers GlobalCooldown matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers Guard matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers HealingDone matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Healing"
            src="/img/healing.png"
            style={
              Object {
                "border": 0,
              }
            }
          />
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Healing done
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Total healing done: <b>1,624,806</b>"
            >
              5,285 HPS
            </dfn>
          </div>
        </div>
      </div>
    </div>
    <div
      className="panel-footer"
      style={
        Object {
          "borderTop": 0,
          "overflow": "hidden",
          "padding": 0,
        }
      }
    >
      <div
        className="statistic-bar"
      >
        <div
          className="stat-healing-bg"
          data-tip="Regular healing done, not including absorbs. You did a total of 1.44m regular healing (74.65% of raw healing)."
          style={
            Object {
              "width": "74.64725935818505%",
            }
          }
        >
          <img
            alt="Healing"
            src="/img/healing.png"
          />
        </div>
        <div
          className="stat-healing-absorbs-bg"
          data-tip="Healing by absorbs. This only includes damage prevented by absorbs. You did a total of 97k healing with absorbs (5.02% of raw healing)."
          style={
            Object {
              "width": "5.020635049532912%",
            }
          }
        >
          <img
            alt="Absorb Healing"
            src="/img/absorbed.png"
          />
        </div>
        <div
          className="stat-negative-absorbs-bg"
          data-tip="Absorbed healing. This only includes healing that gets absorbed by debuffs. You had a total of 85k healing absorbed (4.37% of raw healing)."
          style={
            Object {
              "width": "4.37259259910591%",
            }
          }
        >
          <img
            alt="Absorbed Healing"
            src="/img/negative-absorbed.png"
          />
        </div>
        <div
          className="remainder stat-overhealing-bg"
          data-tip="Overhealing. You did a total of 309k overhealing (15.96 % of raw healing). Overhealing can be caused by playing poorly such as selecting the wrong targets or casting abilities at the wrong time, other healers sniping, and/or bringing too many healers."
        >
          <img
            alt="Overhealing"
            src="/img/overhealing.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers HealingReceived matches the statistic snapshot 1`] = `"module inactive"`;

exports[`The Brewmaster Analyzer analyzers HighTolerance matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=196737"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/monk_ability_avertharm.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Avg. Haste from High Tolerance
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="You spent: <ul>
              <li><b>81s</b> (26.25%) without Stagger.</li>
              <li><b>108s</b> (35.23%) in Light Stagger.</li>
              <li><b>97s</b> (31.66%) in Moderate Stagger.</li>
              <li><b>21s</b> (6.87%) in Heavy Stagger.</li>
            </ul>"
            >
              7.65%
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers IronSkinBrew matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=115308"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_monk_ironskinbrew.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Hits Mitigated w/ Ironskin Brew
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Ironskin Brew breakdown (these values are direct damage and does not include damage added to stagger):
            <ul>
                <li>You were hit <b>137</b> times with your Ironskin Brew buff (<b>2,408,958</b> damage).</li>
                <li>You were hit <b>1</b> times <b><i>without</i></b> your Ironskin Brew buff (<b>15,189</b> damage).</li>
            </ul>
            <b>99.28%</b> of attacks were mitigated with Ironskin Brew.<br/>
            <b>90.42%</b> uptime on the Ironskin Brew buff."
            >
              99.28%
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers IronskinBrew matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers KegSmash matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers MasteryValue matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=117906"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_monk_shuffle.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Expected Mitigation by Mastery
          </div>
          <div
            className="value"
          >
            Click to load
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers MitigationCheck matches the statistic snapshot 1`] = `null`;

exports[`The Brewmaster Analyzer analyzers PurifyingBrew matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=119582"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/inv_misc_beer_06.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Avg. Mitigation per Purifying Brew
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Purifying Brew removed <b>387k</b> damage in total over 6 casts.<br/>
                  The smallest purify removed <b>13k</b> and the largest purify removed <b>95k</b>.<br/>
                  You purified <b>2</b> (33.33%) times without reaching Heavy Stagger.<br/>
                  Your purifies were delayed from the nearest peak by <b>2.19s</b> on average."
            >
              64k
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers RushingJadeWind matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers SharedBrews matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers Stagger matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="panel statistic-box expandable"
  >
    <div
      className="panel-body"
    >
      <div
        className="flex"
      >
        <div
          className="flex-sub statistic-icon"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <a
            href="http://wowhead.com/spell=115308"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            <img
              alt=""
              className="icon"
              src="//render-us.worldofwarcraft.com/icons/56/ability_monk_ironskinbrew.jpg"
            />
          </a>
        </div>
        <div
          className="flex-main"
          style={
            Object {
              "paddingLeft": 16,
              "position": "relative",
            }
          }
        >
          <div
            className="slabel"
          >
            Damage staggered
          </div>
          <div
            className="value"
          >
            <dfn
              data-tip="Incoming damage added to stagger:
          <ul>
            <li>Total physical damage added to stagger: 1,452,326</li>
            <li>Total magical damage added to stagger: 855,778</li>
          </ul>
          Damage taken from stagger:
          <ul>
            <li>Total damage from stagger dot: 1,736,876 (75.25% of total staggered)</li>
            <li>Total damage removed from stagger dot before damaging you: 446,937 (19.36% of total staggered)</li>
          </ul>
        "
            >
              2.31m 
            </dfn>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers StaggerFabricator matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers StaggerPoolGraph matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers StaggeringStrikes matches the statistic snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-sm-6 col-xs-12"
>
  <div
    className="panel statistic-box item"
  >
    <div
      className="panel-body flex"
    >
      <div
        className="flex-sub statistic-icon"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <a
          href="http://wowhead.com/spell=273464"
          rel="noopener noreferrer"
          target="_blank"
        >
           
          <img
            alt=""
            className="icon"
            src="//render-us.worldofwarcraft.com/icons/56/ability_monk_blackoutstrike.jpg"
          />
        </a>
      </div>
      <div
        className="flex-main"
      >
        <div
          className="slabel"
        >
          <a
            href="http://wowhead.com/spell=273464"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            Staggering Strikes
          </a>
        </div>
        <dfn
          className="value"
          data-tip="Your Blackout Strike casts each remove 1,877 staggered damage.
            
            A total of 34 casts removed 60k staggered damage (3,754 overhealed)."
        >
          <img
            alt="Healing"
            className="icon"
            src="/img/healing.png"
          />
           
          3.70 % / 195 HPS
        </dfn>
      </div>
    </div>
  </div>
</div>
`;

exports[`The Brewmaster Analyzer analyzers TigerPalm matches the statistic snapshot 1`] = `"module has no statistic method"`;

exports[`The Brewmaster Analyzer analyzers TrainingOfNiuzao matches the statistic snapshot 1`] = `"module inactive"`;
